diff --git a/build/params_2k.go b/build/params_2k.go
index df1b1bedb..10b9f4fed 100644
--- a/build/params_2k.go
+++ b/build/params_2k.go
@@ -65,7 +65,7 @@ var UpgradeLightningHeight = abi.ChainEpoch(-22)
 
 var UpgradeThunderHeight = abi.ChainEpoch(-23)
 
-var UpgradeWatermelonHeight = abi.ChainEpoch(200)
+var UpgradeWatermelonHeight = abi.ChainEpoch(5)
 
 // This fix upgrade only ran on calibrationnet
 const UpgradeWatermelonFixHeight = -100
